src/adagrams.js

```diff
@@ -1,15 +1,94 @@
 export const drawLetters = () => {
-  // Implement this method for wave 1
+  var letters = {
+    'A':9, 'B':2, 'C':2, 'D':4, 'E':12, 'F':2, 'G':3, 'H':2,
+    'I': 9, 'J': 1, 'K': 1, 'L': 4, 'M': 2, 'N': 6, 'O': 8, 'P': 2, 
+    'Q': 1, 'R': 6, 'S': 4, 'T': 6, 'U': 4, 'V': 2, 'W': 2, 'X': 1, 
+    'Y': 2, 'Z': 1
+  };
+  let pieces = [];
+  let count = 98;
+
+  for(let i = 0; i < 10; i++){
+    let randPiece = Math.floor(Math.random() * count+1);
+    let key = weighted(letters, randPiece);
+    pieces.push(key);
+    
+    letters[key] -=1;
+    if (letters[key] == 0){
+        delete letters[key];
+    }
+    count -= 1;
+  }
+  return pieces;
+};
+
+const weighted = (dict, randomPiece) => {
+  for (let [key, value] of Object.entries(dict)){
+    randomPiece -= value;
+    if (randomPiece <= 0){
+        return key;
+    }
+  }
```

Comment:
Love this function! For readability we want to make sure we are placing our function definitions above the calls.